templates/test/ci/prow-topology/base.yaml

```diff
@@ -0,0 +1,258 @@
+              allocate-node-cidrs: "false"
+              cloud-config: /etc/kubernetes/azure.json
+              cloud-provider: azure
+            extraVolumes:
+              - hostPath: /etc/kubernetes/azure.json
+                mountPath: /etc/kubernetes/azure.json
+                name: cloud-config
+                readOnly: true
+          etcd:
+            local:
+              dataDir: /var/lib/etcddisk/etcd
+              extraArgs:
+                quota-backend-bytes: "8589934592"
+        diskSetup:
+          filesystems:
+            - device: /dev/disk/azure/scsi1/lun0
+              extraOpts:
+                - -E
+                - lazy_itable_init=1,lazy_journal_init=1
+              filesystem: ext4
+              label: etcd_disk
+            - device: ephemeral0.1
+              filesystem: ext4
+              label: ephemeral0
+              replaceFS: ntfs
+          partitions:
+            - device: /dev/disk/azure/scsi1/lun0
+              layout: true
+              overwrite: false
+              tableType: gpt
+        files:
+          - contentFrom:
+              secret:
+                key: control-plane-azure.json
+                name: replace_me
+            owner: root:root
+            path: /etc/kubernetes/azure.json
+            permissions: "0644"
+        initConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              azure-container-registry-config: /etc/kubernetes/azure.json
+              cloud-config: /etc/kubernetes/azure.json
+              cloud-provider: azure
+            name: '{{ ds.meta_data["local_hostname"] }}'
+        joinConfiguration:
+          nodeRegistration:
+            kubeletExtraArgs:
+              azure-container-registry-config: /etc/kubernetes/azure.json
+              cloud-config: /etc/kubernetes/azure.json
+              cloud-provider: azure
+            name: '{{ ds.meta_data["local_hostname"] }}'
+        mounts:
+          - - LABEL=etcd_disk
+            - /var/lib/etcddisk
+        postKubeadmCommands: [ ]
+        preKubeadmCommands: [ ]
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureClusterTemplate
+metadata:
+  name: ci-default-azure-cluster
+spec:
+  template:
+    spec:
+      additionalTags:
+        replace_me_key: replace_me_val
+      identityRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+        kind: AzureClusterIdentity
+      location: replace_me
+      networkSpec:
+        subnets:
+          - role: control-plane
+          - natGateway:
+              name: node-natgateway
+            role: node
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: AzureMachineTemplate
+metadata:
+  name: ci-default-control-plane
+spec:
+  template:
+    spec:
+      dataDisks:
+        - diskSizeGB: 256
+          lun: 0
+          nameSuffix: etcddisk
+      osDisk:
+        diskSizeGB: 128
+        osType: Linux
+      sshPublicKey: ""
+      vmSize: replace_me
```

Comment:
what is this value?

Reply:
This will be patched by the variable passed in cluster definition. Eg. for cp machines, patch: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/48cb6b1bd515fe239f05788d67e6bcaf78e49c6c/templates/test/ci/prow-topology/patches.yaml#L154 ; variable: https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/48cb6b1bd515fe239f05788d67e6bcaf78e49c6c/templates/test/ci/prow-clusterclass/cluster.yaml#L25

Reply:
oh I see its patching this value in and this is a stub for the value 👍